### PR TITLE
34 predefined hints

### PR DESCRIPTION
**Pull request by IssaHanou**
### Issue
Closes GH-34

## Contents
- adds predefined hints to hint menu
- dynamic width of hint boxes
- updates styling on device table

## Screens

Big screen:
image

small screen1:
image

small screen2:
image

dropdown:
image

### CODE REVIEW

**Comment by GwennanS** [comment]
This also needs to work when the back-end is started after the front-end

**Comment by GwennanS** [comment]
I get duplicate hints when the back-end gets restarted

**Comment by GwennanS** [comment]
core.js:6014 ERROR TypeError: events is not iterable
    at Puzzles.setPuzzles (puzzles.ts:28)
    at AppComponent.processSetUp (app.component.ts:199)
    at AppComponent.processMessage (app.component.ts:184)
    at SafeSubscriber._next (app.component.ts:69)
    at SafeSubscriber.__tryOrUnsub (Subscriber.js:185)
    at SafeSubscriber.next (Subscriber.js:124)
    at Subscriber._next (Subscriber.js:72)
    at Subscriber.next (Subscriber.js:49)
    at RefCountSubscriber._next (Subscriber.js:72)
    at RefCountSubscriber.next (Subscriber.js:49)

File: back-end/src/sciler/communication/communication.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ type Communicator struct {
 func NewCommunicator(host string, port int, topicsOfInterest []string) *Communicator {
 	opts := mqtt.NewClientOptions()
 	opts.AddBroker(fmt.Sprintf("%s://%s:%d", "tcp", host, port))
-	opts.SetClientID("back-end")
+	opts.SetClientID("back-end-issa")
```
Comment (GwennanS) [inline back-end/src/sciler/communication/communication.go]: oops

File: back-end/src/sciler/handler/handler_helper.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+		Type:     "instruction",
+		Contents: instructions,
+	}
+	jsonMessage, _ := json.Marshal(&message)
+	logrus.Infof("sending instruction data to %s: %s", clientID, fmt.Sprint(message.Contents))
+	handler.Communicator.Publish(clientID, string(jsonMessage), 3)
+}
+
+// UpdateStatus is the function to process status messages.
+func (handler *Handler) UpdateStatus(raw Message) {
```
Comment (mrlomar) [inline back-end/src/sciler/handler/handler_helper.go]: Why is this method public?

File: back-end/src/sciler/handler/handler_helper.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+		for _, rule := range rules {
+			if rule.Executed < rule.Limit && rule.Conditions.Resolve(handler.Config) {
+				rule.Execute(handler)
+			}
+		}
+	}
+}
+
+// SendEventStatus sends the status of events to the front-end
+func (handler *Handler) SendEventStatus() {
```
Comment (mrlomar) [inline back-end/src/sciler/handler/handler_helper.go]: Why is this method public?

File: back-end/src/sciler/handler/handler_helper.go [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+	"github.com/sirupsen/logrus"
+	"reflect"
+	"sciler/config"
+	"time"
+)
+
+// SendSetUp sends the general set-up information to the front-end.
+// This includes the name, all hints and event descriptions
+// Statuses are also sent
+func (handler *Handler) SendSetUp() {
```
Comment (mrlomar) [inline back-end/src/sciler/handler/handler_helper.go]: should 'SendSetUp' not be 'SendSet**u**p?

**Comment by IssaHanou** [comment]
> 
> 
> Maybe an idea: set a character limit (in the front-end) on the number of characters for a custom hint. This will prevent the site from rendering very slowly as well as displaying a massive table :)

It'd be a good idea, but setting a random value doesn't seem like a good option, so after some manual testing we can add this later

**Pull request merged**